tests/test_optional.py

```diff
@@ -0,0 +1,30 @@
+from typing import Any
+
+from pytest import mark, param, raises
+
+from omegaconf import OmegaConf
+from omegaconf.errors import ValidationError
+from tests import SubscriptedDict, SubscriptedList
+
+
+@mark.parametrize(
+    ("cls,key,assignment,error"),
+    [
+        param(SubscriptedList, "list", [None], True, id="list_elt"),
```

Comment:
Can you add a test when assigning a *partially* valid list, e.g., `[0, 1, None]`?

Reply:
Done in https://github.com/omry/omegaconf/pull/749/commits/c97b3f16245315b685496348ad5a4160b55c1d5b.